pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -1519,13 +1523,52 @@ def handle_qe_notification(self, release_jira: str, release_name: str, impetus_a
         nightlies_w_pullspecs = nightlies_with_pullspecs(nightlies)
         self._send_release_email(release_name, impetus_advisories, jira_issue_link,
                                  nightlies_w_pullspecs)
+        self._logger.info("Update QE's release tests repo...")
+        self._update_qe_repo(release_name, release_jira, impetus_advisories)
         if not self.runtime.dry_run:
             self._jira_client.assign_to_me(subtask)
             self._jira_client.close_task(subtask)
             self._logger.info("Closed subtask %s", subtask.key)
         else:
             self._logger.info("Would've closed subtask %s", subtask.key)
 
+    @retry(reraise=True, stop=stop_after_attempt(3), wait=wait_fixed(10))
+    def _update_qe_repo(self, release_name: str, release_jira: str, advisories: Dict[str, int]):
+        github_client = Github(os.environ.get("GITHUB_TOKEN"))
+        upstream_repo = github_client.get_repo("openshift/release-tests")
+        fork_repo = github_client.get_repo("openshift-bot/release-tests")
+        update_message = f"Add release {release_name}"
+        major, minor = isolate_major_minor_in_group(self.group)
+        file_path = f"_releases/{major}.{minor}/{major}.{minor}.z.yaml"
+        self._logger.info("Updating QE release repo")
+        # create branch
+        for branch in fork_repo.get_branches():
+            if branch.name == release_name:
+                fork_repo.get_git_ref(f"heads/{release_name}").delete()
+        fork_branch = fork_repo.create_git_ref(f"refs/heads/{release_name}", upstream_repo.get_branch("z-stream").commit.sha)
+        self._logger.info("Created fork branch ref %s", fork_branch.ref)
+        # get release file
+        try:
+            release_content = upstream_repo.get_contents(file_path, ref="z-stream")
+            file = yaml.load(release_content.decoded_content, Loader=yaml.FullLoader)
+            file['releases'][release_name] = {'advisories': advisories, 'release_jira': release_jira}
+        except ParserError:
+            self._logger.warning("release file not in valid yaml format, overwrite with new value")
+            file = {'releases': {}}
+            file['releases'][release_name] = {'advisories': advisories, 'release_jira': release_jira}
+        except GithubException:
+            self._logger.warning("release file not found in upstream repo, skip update qe repo")
+            return
+        # update release file
+        file_content_bytes = yaml.dump(file).encode()
+        fork_file = fork_repo.get_contents(file_path, ref=release_name)
+        fork_repo.update_file(file_path, update_message, file_content_bytes, fork_file.sha, branch=release_name)
+        # create pr
+        pr = upstream_repo.create_pull(title=update_message, body=update_message, base="z-stream", head=f"openshift-bot:{release_name}")
+        pr.add_to_labels("lgtm", "approved")
+        pr.merge()
```

Comment:
Can we wrap this and other interactions with GitHub in `try`? Let's not have the promote job fail if this fails.

Reply:
updated in https://github.com/openshift-eng/art-tools/pull/672/commits/995384aa85e26ab602932b4d690eec6b3f944a3c